examples/kuadrant/main.go

```diff
@@ -70,32 +75,82 @@ func main() {
 			schema.GroupKind{Group: kuadrantv1beta3.SchemeGroupVersion.Group, Kind: "AuthPolicy"},
 			schema.GroupKind{Group: kuadrantv1beta3.SchemeGroupVersion.Group, Kind: "RateLimitPolicy"},
 		),
-		controller.WithCallback(buildReconciler(gatewayProviders, client).Reconcile),
+		controller.WithCallback(buildReconciler(gatewayProviders, client)),
 	}
 	controllerOpts = append(controllerOpts, controllerOptionsFor(gatewayProviders)...)
 
 	controller.NewController(controllerOpts...).Start()
 }
 
-func buildReconciler(gatewayProviders []string, client *dynamic.DynamicClient) *Reconciler {
-	var providers []GatewayProvider
+// buildReconciler builds a reconciler that executes the following workflow:
+//  1. log event
+//  2. save topology to file
+//  3. effective policies
+//  4. (gateway deleted) delete SecurityPolicy / (other events) reconcile SecurityPolicies
+//  4. (gateway deleted) delete AuthorizationPolicy / (other events) reconcile AuthorizationPolicies
+func buildReconciler(gatewayProviders []string, client *dynamic.DynamicClient) controller.CallbackFunc {
+	effectivePolicyReconciler := &reconcilers.EffectivePoliciesReconciler{Client: client}
+
+	commonAuthPolicyResourceEventMatchers := []controller.ResourceEventMatcher{
+		{Resource: ptr.To(gwapiv1.SchemeGroupVersion.WithResource("gatewayclasses"))},
+		{Resource: ptr.To(gwapiv1.SchemeGroupVersion.WithResource("gateways")), EventType: ptr.To(controller.CreateEvent)},
+		{Resource: ptr.To(gwapiv1.SchemeGroupVersion.WithResource("gateways")), EventType: ptr.To(controller.UpdateEvent)},
+		{Resource: ptr.To(gwapiv1.SchemeGroupVersion.WithResource("httproute"))},
```

Comment:
```suggestion
		{Resource: ptr.To(gwapiv1.SchemeGroupVersion.WithResource("httproutes"))},
```
We're not correctly reconciling to `HTTPRoute` events due to typo, which preventing matching with the reconcile function that should be run

Reply:
Might be good to put these `GroupVersionResources` as vars that can be re-used as they are used in many places

Reply:
Sharp eyes! Thanks!

Reply:
Done.

Please let me know what you think about the files chosen to store the global variables. I wasn't sure about the policy API type ones in particular, but you'll tell me.